src/main/java/org/springframework/cloud/stream/app/plugin/ScsProjectGenerator.java

```diff
@@ -37,6 +38,13 @@
 
     @Override
     protected File doGenerateProjectStructure(ProjectRequest request) {
+        return doGenerateProjectStructure(request, MavenModelUtils.ENTRYPOINT_TYPE_EXEC);
+    }
+
+    protected File doGenerateProjectStructure(ProjectRequest request, String entrypointType) {
+        if (StringUtils.isEmpty(entrypointType)) {
```

Comment:
it would probably be good to do some validation here so if an `entrypointType` is provided we validate its one we support. further down in the changes theres a check to see if its `shell`, if not fall back to `exec` but theres really no indication to the user they may be falling back to `exec` due to a typo or whatnot..